windows/README_MINGW

```diff
@@ -0,0 +1,218 @@
+script will later check if the SDK is present at /opt/asiosdk
+
+wget https://www.steinberg.net/asiosdk -O /tmp/asiosdk.zip
+unzip /tmp/asiosdk.zip -d /tmp
+mv /tmp/asiosdk_{version}_{date} /opt/asiosdk
+
+The PKGBUILD description file needs to be patched so configure is called with
+the necessary flags for ASIO:
+
+--with-asiodir=/opt/asiosdk   <- new option     
+--with-winapi=wmme,directx,wasapi,vdmks,asio   <- append 'asio' to existing
+
+Currently the shared library version of portaudio fails to build, a further
+patch is needed so only the static version is generated. Comment out lines
+[61-75], [82-83] from PKGBUILD.
+
+
+
+Compiling JACK
+--------------
+
+- Clone repo
+
+git clone https://github.com/jackaudio/jack2
+
+- Build and install
+
+cd jack2
+./waf configure --prefix=/opt/jack && ./waf -p install
+
+The resulting files can be found at /c/opt/jack/bin, or c:\msys64\opt\jack\bin
+
+
+
+Compiling JackRouter
+--------------------
+
+Assuming you have Visual Studio installed, just open and build the project:
+
+windows/JackRouter/JackRouter_MinGW_deps.vcxproj
+
+Differences from the original JackRouter.vcxproj:
+
+- Depends on the asiosdk files from the previous MinGW JACK installation (i.e.,
+  it points to absolute paths starting with c:\msys64\opt)
+- Links to the jack.dll.a created by the previous build, instead of libjack.lib
+  This means JackRouter.dll will currently depend on libjack-0.dll
+- Force includes stdint.h and defines _STDIN_H to avoid int8_t, int32_t and
+  uint32_t basic types redefinition duringn compilation
+
+The project was successfully tested to build a 64bits version of JackRouter.dll
+using Visual Studio 2017.
+
+
+
+Running and distributing
+------------------------
+
+Make sure the following DLLs are distributed together with the JACK binaries.
+They can be found in c:\msys64\mingw64\bin:
+
+libstdc++-6.dll
+libgcc_s_sjlj-1.dll
+libwinpthread-1.dll
+libtre-5.dll
+libsystre-0.dll
+libdb-6.0.dll
+libsndfile-1.dll
+libsamplerate-0.dll
+
+To enable JACK-enabled applications connect to the freshly built server, copy
+the following DLLs to c:\windows (that is what the current official 1.9.11
+installer for Windows does, needs to be improved):
+
+libjackserver-0.dll   (rename to libjackserver64.dll)
+libjack-0.dll         (rename to libjack64.dll)
+libgcc_s_sjlj-1.dll
+libwinpthread-1.dll
+libtre-5.dll
+libsystre-0.dll
+
+JackRouter.dll is created inside windows/JackRouter/Release64. It depends on
+libjack through libjack-0.dll instead of libjack64.dll. To avoid polluting
+c:\windows even more, the dependency dll can be copied to the same directory
+where JackRouter.dll will reside. Once done it needs to be registered:
+
+regsvr32 JackRouter.dll
+
+Tested working clients:
+Ardour
+Bitwig Studio
+QJackCtl
+JackRouter
+
+Example of starting the JACK server including MIDI support for a Focusrite USB
+audio device using ASIO:
+jackd -R -S -X winmme -d portaudio -p 32 -r 48000 -d "ASIO::Focusrite USB ASIO"
```

Comment:
My jack setup, which is working fine with 1.9.11, crashes upon starting: 
```
08:47:44.103 Statistics reset.
08:47:55.518 JACK is starting...
08:47:55.519 ./jackd.exe -S -dportaudio -r48000 -p1024 -D -C"ASIO::ASIO HDSPe FX" -P"ASIO::ASIO HDSPe FX"
08:47:55.573 JACK was started with PID=78262880.
08:47:55.586 JACK was stopped
08:47:55.587 JACK has crashed.
08:47:57.723 Could not connect to JACK server as client. - Overall operation failed. Please check the messages window for more info.
```
The issue might be that the instructions from line 168-178 didn't exactly tell me where to put the files that were not mentioned in lines 184-189. So where to put libsndfile-1, libsamplerate-0 and libdb-6.0?

Reply:
You can place the DLLs listed in 168-178 in the same directory the binaries will finally reside in your disk, for example `C:\Program Files\Jack`. Windows default behavior is to first look for DLLs in the same directory the invoked EXEs are located, then in system-wide directories like `C:\Windows`, `C:\Windows\System32`, and a couple more. The guide does not dig too much into deployment details because this should be either addressed by a waf script update, or better a wizard-like installer which will be more familiar to Windows users, like the current binary distribution implements. There is a recommendation to place the libraries listed in 184-189 in `C:\Windows` because those are dependencies of `libjack-0.dll` _only_, and any Jack client will indirectly need to link against them (in contrast the first DLL list is for `jackd.exe` et al.) . It could be any other system-wide DLLs directory, I just followed what the current binary distribution does. Do not forget to rename `libjack-0.dll` to `libjack64.dll` inside `C:\Windows` otherwise clients will not be able to find it.

I think the guide can be improved by adding an example of how to install everything to Program Files, until an automated solution is developed.

If it continues to crash then something else is happening. Btw, where did you take the log excerpt from?

Reply:
> 
> 
> You can place the DLLs listed in 168-178 in the same directory the binaries will finally reside in your disk, for example `C:\Program Files\Jack`. 

Cool, that works. Awesome! I can also confirm that this works with the latest VS2019 and toolchain+SDK.
I know about the default behaviour, it just wasn't clear from the description in I had less time for trying things out yesterday. You might want to include that info.

I'm surprised that the libraries from 171-178 are different from line 184-189. Any reason why? I'm working on a jack client for unity. With the binaries generated from this guide, that jack client only works if I copy over all of the dlls from line 171-178 to `c:\windows` while jack-enabled applications should just need the dlls from line 184-189. My jack client works fine using the 1.9.11 binary release.

Reply:
Great to know it also works with VS2019. I updated the guide to reflect that and clarify the paths selection. I am not really familiar with the JACK internals, but think libjack only needs a subset of the server dependencies, for example it does not need to interface to the audio device hence portaudio is not needed by libjack. That explains the discrepancies between DLL lists. I am also not a proficient Windows programmer, but know that cluttering the Windows directory is not a friendly thing to do. I think libjack should be either statically linked to its dependencies or moved to its own place and made available to clients somehow (if that is possible at all). Then the server + standalone binaries + dependency DLLs are not a problem if placed in their own directory. The updated guide shows how to do that using` C:\Program Files\Jack` as an example.

Thanks for all the contributions, I am pushing the updated guide now.